### ISSUE TITLE
Increase the maximum size limit

### ISSUE THREAD

**Issue by YePpHa** [OpenUserJs/OpenUserJS.org#360]
I am the developer of YouTube Center and when I'm trying to update YouTube Center on OpenUserJS  I'm met with a problem, which is caused by the maximum request size of the browser (Error: `Request Entity Too Large`).

The first time a userscript is created on OpenUserJS you have the ability to upload the userscript instead of copy/pasting it into a text field. I would request that this ability to upload the file will be added to the page when updating a userscript as YouTube Center minified has a size of ~2.3 MB (I know it's a lot, but it's also a big project).

Also I would probably be able to implement this feature into OpenUserJS myself, but I will just post this here before I begin to do a pull request.

**Comment by qsniyg** [comment]
Would it be possible to have a case-by-case basis for the maximum script size? My userscript (which is [currently hosted on greasyfork](https://greasyfork.org/en/scripts/36662-image-max-url/code)) currently sits at almost 3MB with comments, ~1.2MB without comments. It doesn't host any resources or any libraries, the size primarily comes from the ~3200 rules that exist in the script. Currently this fits under Greasyfork's 2MB limit, but I'm worried that within a year or so, it will surpass the limit as I find and add new rules to the script.

Using `@require` would _technically_ work, but it would require creating one or more generally useless "libraries" (only useful to this userscript), that wouldn't alleviate server or user bandwidth in any way. It'd also make it harder for users (as well as myself) to modify the userscript and understand what's going on, and further complicate the rest of the project's uses.

**Comment by Martii** [comment]
@qsniyg 

Presuming you don't have greater than four spaces in between JavaScript quotes/apostrophes/back-ticks *(String Literals)* that you need to keep, and Linux or compatible conversion *(or and editor that will do it for you)*... I get this for your script when converting spaces to tabs **keeping comments**:

``` console
$ sed -e 's/    /\t/g' 'Image Max URL.user.js' > "Image Max URL.tabbed.user.js"
$ ls -l Image\ Max\ URL.*
*clipped*  915407 Sep 24 20:29 'Image Max URL.tabbed.user.js'  # 0.873000145 MiB
*clipped* 1162418 Sep 24 19:32 'Image Max URL.user.js'         # 1.108568192 MiB
```

One note on publishing to OUJS if you import from GH  it should be able to be within the limits since it's a direct copy instead of Ace adjusted or run into #793 which is briefly alluded to [up here](https://github.com/OpenUserJS/OpenUserJS.org/issues/360#issuecomment-59456392).

In short ... probably not, and it took about ~3-4 seconds for me to upload via file, but you have options atm.

You will also need to change the `@license` to `Apache-2.0` for SPDX compliance and `@icon` is way too large atm *(640x640 px units currently and max is 256x256 px units for rastered.... do a SVG?)*... so the server will reject your script.

> ... sits at almost 3MB with comments

Not sure where you are getting this number from.

**Comment by qsniyg** [comment]
Alright, good to know, and thank you very much for the ideas, I hadn't even thought about spaces->tabs! :)
